src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -356,6 +398,19 @@ public String supportedKeyType() {
             return SUPPORT_KEY_TYPE;
         }
 
+        @Override
+        public X509Certificate getCertificate(URI privateKeyUri, URI certificateUri) throws KeyLoadingException {
+            KeyManager[] km = getKeyManagers(privateKeyUri, certificateUri);
+            if (km.length == 1 && km[0] instanceof X509KeyManager) {
+                X509KeyManager x509KeyManager = (X509KeyManager) km[0];
+                X509Certificate[] certChain = x509KeyManager.getCertificateChain(KEY_ALIAS);
+                if (certChain.length >= 1) {
+                    return certChain[0];
+                }
+            }
+            return null;
```

Comment:
this is totally unnecessary.

`EncryptionUtils.loadX509Certificates(Paths.get(certificateUri));`

Reply:
Ok, that's better.